### DISCUSSION TITLE
How to customize the "clr-control-error"?

### DISCUSSION THREAD

**Discussion by yandong01** [vmware/clarity#5582] [Q&A]
It seems the mark "clr-control-error" only works for input box with 'required', e.g:
`<input placeholder="Enter your last name" clrInput [(ngModel)]="lastName" name="lastname" required />
    <clr-control-error>You must provide your last name!</clr-control-error>`

I want to know if there is a way to make it works for other error types, e.g:
the username exceed the limit for the input box or the username contains illegal characters.

**Comment by gnomeontherun** [comment]
Take a look at https://clarity.design/angular-components/form/#helper-and-validation-messages for details how you can make the error messages use specific error states depending on the validators. You can use more than one if you want to have different messages. You just pass the string name of the error type to `clrIfError` like so.

```<clr-control-error *clrIfError="'required''">Only show this message when required error</clr-control-error>
<clr-control-error *clrIfError="'maxlength''">Only show this message when maxlength error</clr-control-error>
```

**Reply by yandong01** [reply]
Thanks. But how to use this "pattern", could you give me an example, i wanted to exclude some characters in my input box.

I tried below but not work.
` <input pattern="[^a-z]"/>`

**Reply by gnomeontherun** [reply]
If you inspect the validators in Angular to find the name, I believe it is just `pattern`, you will use the name from the ngControl errors array. `*clrIfError="'pattern'"`
